From 3ccd25c981431426038d7952f5b0b86118d92c23 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Sat, 21 Jan 2017 05:57:17 -0500
Subject: [PATCH 2/2] RHEL 5 ONLY QEMU-NBD 1.4 HAS NO -f OPTION

---
 p2v/nbd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/p2v/nbd.c b/p2v/nbd.c
index b1caf2f..2232b08 100644
--- a/p2v/nbd.c
+++ b/p2v/nbd.c
@@ -409,7 +409,7 @@ start_qemu_nbd (const char *device,
               "-r",            /* readonly (vital!) */
               "-p", port_str,  /* listening port */
               "-t",            /* persistent */
-              "-f", "raw",     /* force raw format */
+              //"-f", "raw",     /* force raw format */
               "-b", ipaddr,    /* listen only on loopback interface */
               "--cache=unsafe",  /* use unsafe caching for speed */
               device,            /* a device like /dev/sda */
@@ -424,7 +424,7 @@ start_qemu_nbd (const char *device,
               "qemu-nbd",
               "-r",            /* readonly (vital!) */
               "-t",            /* persistent */
-              "-f", "raw",     /* force raw format */
+              //"-f", "raw",     /* force raw format */
               "--cache=unsafe",  /* use unsafe caching for speed */
               device,            /* a device like /dev/sda */
               NULL);
-- 
1.8.2.3

