From 28dd464c8e78f241622d142671a61a75bf5d758e Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Sat, 21 Jan 2017 05:30:40 -0500
Subject: [PATCH 1/2] RHEL 5 ONLY DISABLE AUTOMATIC REMOTE PORT ALLOCATION

---
 p2v/ssh.c | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/p2v/ssh.c b/p2v/ssh.c
index 8beaf74..919f2df 100644
--- a/p2v/ssh.c
+++ b/p2v/ssh.c
@@ -1044,16 +1044,28 @@ open_data_connection (struct config *config,
     "-N",
     NULL
   };
+#if 0
   CLEANUP_FREE char *port_str = NULL;
   const int ovecsize = 12;
   int ovector[ovecsize];
+#endif
 
-  snprintf (remote_arg, sizeof remote_arg, "0:%s:%d", local_ipaddr, local_port);
+  /* RHEL 5 hack: ssh does not print the "Allocated port ..."  string,
+   * so we cannot find the remotely allocated port.  Instead just
+   * assign a random port and hope for the best.
+   */
+  static int next_remote_port = 58123;
+
+  snprintf (remote_arg, sizeof remote_arg, "%d:%s:%d",
+	    next_remote_port, local_ipaddr, local_port);
+  *remote_port = next_remote_port;
+  next_remote_port++;
 
   h = start_ssh (0, config, (char **) extra_args, 0);
   if (h == NULL)
     return NULL;
 
+#if 0
   switch (mexp_expect (h,
                        (mexp_regexp[]) {
                          { 100, .re = portfwd_re },
@@ -1094,6 +1106,7 @@ open_data_connection (struct config *config,
     mexp_close (h);
     return NULL;
   }
+#endif
 
   return h;
 }
-- 
1.8.2.3

